client/src/Components/CarData.js

```diff
@@ -15,17 +15,16 @@ import FuelEconomyCard from './Card/FuelEconomyCard';
 import GoogleMap from './GoogleMap/GoogleMap';
 import AntdTableComponent from './AntdTable/AntdTableComponent';
 
-export default function CarData() {
+export default function CarData(props) {
   const { loading, error, data } = useQuery(getDashboardAllDataQuery, {
     variables: {
-      vehicleId: `27e1dabc-a89a-444d-bcc3-ace5a33a3d26`,
+      vehicleId: `${props.state}`,
```

Comment:
state is not a good prop name in my opinion. I would suggest props.vehicleId

Reply:
changed to vehicleId